Feature/lp 109 ProfileViewModel

JoseGR98 opened the pull request:
Profile - ViewModel
[LP-109](https://devaptivist.atlassian.net/browse/LP-109)
View Model implementation for the Profile View.
Files added/edited:
- ProfileView that contains the UI
- ProfileViewModel that handles the logic of data
- ProfileModel that contains the data class for the User Profile Data
NavGraph of Profile edited because of new parameter -  TODO

JoseGR98 committed:
ViewModel tasks progress:
ProfileModel added
ProfileViewModel added & worked
build.gradle (profile) modified

JoseGR98 committed:
Merge branch 'develop' into feature/lp-109-profile-viewmodel

JoseGR98 committed:
Merge branch 'develop' into feature/lp-109-profile-viewmodel

JoseGR98 committed:
ProfileView and ProfileViewModel progress

JoseGR98 committed:
Merge branch 'develop' into feature/lp-109-profile-viewmodel

irv205 commented on core/data/src/main/java/com/iteneum/ProfileModel.kt:
```diff
@@ -0,0 +1,10 @@
+package com.iteneum
+
+data class ProfileModel (
```
Rename "ProfileModel" to "Profile"

JoseGR98 replied:
Sure, no problem. Done!

irv205 commented on core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt:
```diff
@@ -14,7 +14,7 @@ fun NavGraphBuilder.dashboardNavGraph(navController: NavHostController) {
     ) {
         composable(route = DashBoardScreen.Profile.route) {
-            ProfileView()
+            ProfileView(navigation = { /*TODO*/ })
```
Explain in that TODO what you need to do in that specific part.

JoseGR98 replied:
Got it! Done!

irv205 commented on feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt:
```diff
@@ -28,18 +29,30 @@ import androidx.compose.ui.res.painterResource
 @Composable
 fun ProfileView(
-    /*TODO expected parameters as navigation and view model*/
+    navigation: () -> Unit /* TODO - ProfileView - Check navigation parameter*/ ,
```
rename this parameter, it depends what do you want, if it's to navigate back you can rename it as "BackToDashBoard" or if it's to navigate to other view you can rename "navigateToNameView"

JoseGR98 replied:
It is for going back, by pressing the arrow component, but not implemented yet. Changed the name for the one you mentioned. Thank you!

JoseGR98 committed:
LP-109 Progress
DashBoardNavGraph.kt improved
Profile.kt name modified
ProfileView and ProfileViewModel.kt improved

carlosph816 commented on feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt:
```diff
@@ -0,0 +1,99 @@
+
+@HiltViewModel
+class ProfileViewModel @Inject constructor(): ViewModel() {
+    private val _profileModelMutable : MutableState<Profile> = mutableStateOf(Profile())
```
we can change this in the same way you did with profileModel

JoseGR98 replied:
Ok, done!

carlosph816 commented on feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt:
```diff
@@ -28,18 +29,30 @@ import androidx.compose.ui.res.painterResource
 
+@Composable
+fun ProfileContainer(
+    profileViewModel: ProfileViewModel = hiltViewModel()
```
we don't need to give the viewmodel here, you need to implement the hight order functions for events

JoseGR98 replied:
Got it, done!

JoseGR98 committed:
Merge branch 'develop' into feature/lp-109-profile-viewmodel

JoseGR98 committed:
LP-109 Progress. ViewModel improvements:
- UIEventProfile.kt added
- ProfileView.kt modified
- ProfileViewModel.kt modified

JoseGR98 commented:
New PR made. Code review pending

carlosph816 commented on feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt:
```diff
@@ -0,0 +1,95 @@
+                )
+            }
+            is DataState.Loading -> {
+                profileModelMutable = Profile(
```
this shouldn't be here, let's delete and put a "TODO"

JoseGR98 replied:
Got it. Modified!

carlosph816 commented on feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt:
```diff
@@ -0,0 +1,95 @@
+                profileModelMutable = response.data
+            }
+            is DataState.Error -> {
+                profileModelMutable = Profile(
```
this shouldn't be here, let's delete and put a "TODO"

JoseGR98 replied:
Got it. Modified!

JoseGR98 committed:
Merge branch 'develop' into feature/lp-109-profile-viewmodel

JoseGR98 committed:
LP-109 Progress. pfViewModel little Datastate changes.

JoseGR98 commented:
1 code reviewer approved. Waiting for one more approval from you @carlosph816 
Both approved! Only merge left! :D
